Panic with incorrect duration

Hi, devs! I found that when trying to run the restic binary with the --keep-within flag (and other flags with duration) with an incorrect value, it panics.

Output of `restic version`
--------------------------
```
restic 0.14.0 compiled with go1.23.2 on linux/amd64
```

Problem description / Steps to reproduce
----------------------------------------
```
// Example of a flag value that causes panic:
$ xxd testcase.txt
00000000: df80

// Restic's output:
$ ./restic forget --keep-within <data_from_testcase.txt>
panic: strconv.Atoi: parsing "߀": invalid syntax

goroutine 1 [running]:
github.com/restic/restic/internal/restic.nextNumber({0x7ffcc7c1b226?, 0x416a45?})
        /home/fuzzuser/fuzz-projects/restic-0.14.0_fuzz/project/internal/restic/duration.go:69 +0x232
github.com/restic/restic/internal/restic.ParseDuration({0x7ffcc7c1b226?, 0xc000156320?})
        /home/fuzzuser/fuzz-projects/restic-0.14.0_fuzz/project/internal/restic/duration.go:90 +0x6c
github.com/restic/restic/internal/restic.(*Duration).Set(0x19b3090, {0x7ffcc7c1b226?, 0x10ac63e?})
        /home/fuzzuser/fuzz-projects/restic-0.14.0_fuzz/project/internal/restic/duration.go:120 +0x25
github.com/spf13/pflag.(*FlagSet).Set(0xc000016800, {0x10ac63e, 0x12}, {0x7ffcc7c1b226, 0x2})
        /home/fuzzuser/go/pkg/mod/github.com/spf13/pflag@v1.0.5/flag.go:463 +0xa2
github.com/spf13/pflag.(*FlagSet).Parse.func1(0xf0b800?, {0x7ffcc7c1b226?, 0x7ffcc7c1b213?})
        /home/fuzzuser/go/pkg/mod/github.com/spf13/pflag@v1.0.5/flag.go:1138 +0x32
github.com/spf13/pflag.(*FlagSet).parseLongArg(0xc000016800, {0x7ffcc7c1b211, 0x14}, {0xc00059c350, 0x1, 0x1}, 0xc000537bf8)
        /home/fuzzuser/go/pkg/mod/github.com/spf13/pflag@v1.0.5/flag.go:1000 +0x24f
github.com/spf13/pflag.(*FlagSet).parseArgs(0xc000016800, {0xc00059c340?, 0x14?, 0x13?}, 0xc000537bf8)
        /home/fuzzuser/go/pkg/mod/github.com/spf13/pflag@v1.0.5/flag.go:1108 +0x1b1
github.com/spf13/pflag.(*FlagSet).Parse(0xc000016800, {0xc00059c340, 0x2, 0x2})
        /home/fuzzuser/go/pkg/mod/github.com/spf13/pflag@v1.0.5/flag.go:1141 +0xd2
github.com/spf13/cobra.(*Command).ParseFlags(0x19a0020, {0xc00059c340, 0x2, 0x2})
        /home/fuzzuser/go/pkg/mod/github.com/spf13/cobra@v1.5.0/command.go:1655 +0xd2
github.com/spf13/cobra.(*Command).execute(0x19a0020, {0xc00059c340, 0x2, 0x2})
        /home/fuzzuser/go/pkg/mod/github.com/spf13/cobra@v1.5.0/command.go:795 +0x12a
github.com/spf13/cobra.(*Command).ExecuteC(0x19a1e20)
        /home/fuzzuser/go/pkg/mod/github.com/spf13/cobra@v1.5.0/command.go:990 +0x38d
github.com/spf13/cobra.(*Command).Execute(...)
        /home/fuzzuser/go/pkg/mod/github.com/spf13/cobra@v1.5.0/command.go:918
main.main()
        /home/fuzzuser/fuzz-projects/restic-0.14.0_fuzz/project/cmd/restic/main.go:98 +0x165
```

Expected behavior
-----------------
I suppose that this error should probably be handled separately to avoid causing panic.

Actual behavior
---------------
Restic panics.

<img width="1776" height="1102" alt="Image" src="https://github.com/user-attachments/assets/ead6bdac-dd1f-412d-a208-bcf3ee76fb56" />

--- a/internal/restic/duration.go
+++ b/internal/restic/duration.go
@@ -4,7 +4,6 @@ import (
 	"fmt"
 	"strconv"
 	"strings"
-	"unicode"
 
 	"github.com/restic/restic/internal/errors"
 )
@@ -52,7 +51,7 @@ func nextNumber(input string) (num int, rest string, err error) {
 	}
 
 	for i, s := range input {
-		if !unicode.IsNumber(s) {
+		if s < '0' || s > '9' {
 			rest = input[i:]
 			break
 		}

--- a/internal/restic/duration_test.go
+++ b/internal/restic/duration_test.go
@@ -37,6 +37,9 @@ func TestNextNumber(t *testing.T) {
 		{
 			input: "5d  ", num: 5, rest: "d  ",
 		},
+		{
+			input: "5", num: 5, rest: "",
+		},
 	}
 
 	for _, test := range tests {
@@ -78,6 +81,7 @@ func TestParseDuration(t *testing.T) {
 		{input: "2w", err: true},
 		{input: "1y4m3w1d", err: true},
 		{input: "s", err: true},
+		{input: "\xdf\x80", err: true}, // NKO DIGIT ZERO; we want ASCII digits
 	}
 
 	for _, test := range tests {

